tests/core/model_management/test_local_job_predictor.py

```diff
@@ -0,0 +1,304 @@
+# Copyright The Caikit Authors
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""
+Tests for LocalJobPredictor
+"""
+# Standard
+from datetime import timedelta
+from pathlib import Path
+import multiprocessing
+import os
+import tempfile
+import threading
+
+# Third Party
+import pytest
+
+# First Party
+import aconfig
+
+# Local
+from caikit.config import get_config
+from caikit.core import ModuleBase
+from caikit.core.data_model import DataStream, PredictionJobStatus
+from caikit.core.exceptions.caikit_core_exception import CaikitCoreException
+from caikit.core.model_management.local_job_predictor import LocalJobPredictor
+from sample_lib.data_model import SampleInputType, SampleOutputType
+from sample_lib.modules import SampleModule
+
+## Helpers #####################################################################
+
+
+def local_predictor(**kwargs) -> LocalJobPredictor:
+    cfg = aconfig.Config(kwargs, override_env_vars=False)
+    return LocalJobPredictor(cfg, "test-instance")
+
+
+@pytest.fixture
+def save_path():
+    with tempfile.TemporaryDirectory() as workdir:
+        yield os.path.join(workdir, "model_save_dir")
+
+
+@pytest.fixture()
+def predictor_type_cfg(save_path):
+    yield {"result_dir": save_path}
+
+
+class WaitPredict:
+    """Dummy module that will block prediction on an event"""
+
+    def run(self, wait_event: threading.Event) -> SampleOutputType:
+        wait_event.wait()
+        return SampleOutputType(greeting="hello world")
+
+
+## Tests #######################################################################
+
+
+def test_predict_and_get_info(predictor_type_cfg):
+    """Test that running a prediction can fetch status correctly"""
+    predictor = local_predictor(**predictor_type_cfg)
+
+    # Launch the training and force it to wait
+    # NOTE: Data stream passed by positional arg to ensure it is passed through
+    #   correctly by position
+    wait_event = threading.Event()
+    model_future = predictor.predict(
```

Comment:
Another copy-paste NIT, but I'd like to not call these `model_future` to avoid confusion.